webbpsf/webbpsf_core.py

```diff
@@ -1017,7 +1017,15 @@ def calc_psf(self, outfile=None, source=None, nlambda=None, monochromatic=None,
         self.options['add_distortion'] = add_distortion
         self.options['crop_psf'] = crop_psf
 
-        # Run poppy calc_psf
+        # UPDATE THE OPD V2V3 BASED ON DETECTOR POSITION, IN ORDER TO CALCULATE SM FIELD-DEPENDENT WFE.
+        # SEE opds._apply_sm_field_dependence_model()
+        #
+        # v2v3 attribute exists only if using the linear model, so check first:
+        if hasattr(self.pupil, 'v2v3'):
+            self.pupil.v2v3 = self._tel_coords().to(units.arcsec)
+            self.pupil.update_opd()
```

Comment:
We should make this more clever so as to not re-invoke the update_opd machinery for multiple calculations at the same field point. 

```
if hasattr(self.pupil, 'v2v3') and self.pupil.v2v3 != self._tel_coords().to(units.arcsec):
```

Reply:
Done

Reply:
Done.